api/app/models/model.py

```diff
@@ -1,197 +1,156 @@
-                result.append(i)
-    return result
+        connection.commit()
+        status = True
+        return status
 
 
-def record_by_zone(zone):
-    result = list()
+def delete(table, field=None, value=None):
+    cursor, connection = get_db()
+    rows_deleted = 0
     try:
-        data_rec = read_all("record")
-    except Exception:
-        data_rec = []
+        query = f'DELETE FROM "{table}" WHERE {field}={value}'
+        cursor.execute(query)
+    except (Exception, psycopg2.DatabaseError) as error:
+        connection.rollback()
+        raise error
     else:
-        for i in data_rec:
-            data = None
-            if i['zone'] == zone:
-                type_data = read_by_id("type", i['type'])
-                ttl_data = read_by_id("ttl", i['ttl'])
-                try:
-                    content_data = content_by_record(i['key'])
-                except Exception:
-                    content_data = []
-                if i['serial']:
-                    try:
-                        serial_data = serial_by_record(i['key'])
-                    except Exception:
-                        data = {
-                            "key": i['key'],
-                            "value": i['value'],
-                            "created_at": i['created_at'],
-                            "type": type_data,
-                            "ttl": ttl_data,
-                            "content": content_data,
-                            "serial": []
-                        }
-                    else:
-                        data = {
-                            "key": i['key'],
-                            "value": i['value'],
-                            "created_at": i['created_at'],
-                            "type": type_data,
-                            "ttl": ttl_data,
-                            "content": content_data,
-                            "serial": serial_data
-                        }
-                else:
-                    data = {
-                        "key": i['key'],
-                        "value": i['value'],
-                        "created_at": i['created_at'],
-                        "type": type_data,
-                        "ttl": ttl_data,
-                        "content": content_data
-                    }
-            if data is not None:
-                result.append(data)
-        return result
-
-
-def record_delete(key):
-    try:
-        record_data = read_by_id("record", key)
-    except Exception as e:
-        raise e
+        connection.commit()
+        rows_deleted = cursor.rowcount
+        return str(rows_deleted)
+
+
+def retry_execute(query, column, retry_counter, error):
+    limit_retries = 5
+    results = []
+    cursor, connection = get_db()
+    if retry_counter >= limit_retries:
+        raise error
     else:
+        retry_counter += 1
+        print("got error {}. retrying {}".format(str(error).strip(), retry_counter))
```

Comment:
can you use logger exception for std error msg? https://docs.python.org/3/library/logging.html#logging.Logger.exception